### PR TITLE
[14_0] Change CICADA model to be function local

### PR DESCRIPTION
**Pull request by aloeliger**
#### PR description:

Taken from https://github.com/cms-sw/cmssw/pull/44436.

> This PR changes the L1T CICADA function model to be local the function where it is used, where it is hoped that the potential bug seen in https://github.com/cms-sw/cmssw/issues/44435, https://github.com/cms-sw/cmssw/pull/44397, https://github.com/cms-sw/cmssw/pull/44433. No change is expected to CICADA, however it is hoped that Trigger changes for AXOL1Tl might be seen (becoming more reasonable and removing the possible collision issue).

@mmusich & @artlbv 

#### PR validation:

Tests in https://github.com/cms-sw/cmssw/pull/44436. reveal that AXO no-longer triggers all events in 14_1

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This PR is a backport to 14_0

### CODE REVIEW

**Comment by aloeliger** [comment]
backport of https://github.com/cms-sw/cmssw/pull/44436

**Comment by aloeliger** [comment]
please test

File: L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc [hunk 1]
```diff
@@ -109,11 +109,9 @@ class L1TCaloSummary : public edm::stream::EDProducer<> {
 
   UCTLayer1* layer1;
 
-  hls4mlEmulator::ModelLoader loader;
-  std::shared_ptr<hls4mlEmulator::Model> model;
-
   bool overwriteWithTestPatterns;
   std::vector<edm::ParameterSet> testPatterns;
+  std::string modelName;
```
Comment (fwyzard) [inline L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc:114]: Please rename all data members to follow CMS conventions: [2.9](Start data member names with lower case. A trailing “_” is the preferred method to distinguish a data member from the getter method (e.g. momentum_).):

> Start data member names with lower case. A trailing `_` is the preferred method to distinguish a data member from the getter method (e.g. `momentum_`).

File: L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc [hunk 2]
```diff
@@ -184,6 +180,11 @@ void L1TCaloSummary<INPUT, OUTPUT>::produce(edm::Event& iEvent, const edm::Event
 
   UCTGeometry g;
 
+  hls4mlEmulator::ModelLoader loader(modelName);
+  std::shared_ptr<hls4mlEmulator::Model> model;
+  //anomaly trigger loading
+  model = loader.load_model();
```
Comment (fwyzard) [inline L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc:186]: what is the impact of the job timing / throughput of creating an `hls4mlEmulator::ModelLoader` object and loading the model for every event ?

**Pull request closed without merging**